src/connector/src/source/nats/source/message.rs

```diff
@@ -30,7 +31,10 @@ impl From<NatsMessage> for SourceMessage {
             key: None,
             payload: Some(message.payload),
             // For nats jetstream, use sequence id as offset
-            offset: message.sequence_number,
+            //
+            // DEPRECATED: no longer use sequence id as offset, let nats broker handle failover
+            // use reply_subject as offset for ack use, we just check the persisted state for whether this is the first run
+            offset: message.reply_subject.unwrap_or_default(),
```

Comment:
@yufansong Unfortunately, `offset` column should be kept as additional column and I think it is not worth making an exception for Nats. We use `offset` here to store the reply subject.

Reply:
Get it. SGTM